### DISCUSSION TITLE
Unable to build WiX develop (v5 preview) with latest VS 2022

### DISCUSSION THREAD

**Discussion by chrpai** [wixtoolset/issues#7892] [WiX Development]
I was researching cleaning WiX on a clean VM so that I could update the README.md but I've hit a blocker that I don't know how to solve.  When running devbuild.cmd it fails and subsequently stops.

C:\Program Files\dotnet\sdk\8.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(251
,5): error NETSDK1201: For projects targeting .NET 8.0 and higher, specifying a RuntimeIdentifier will no longer produc
e a self contained app by default. To continue building self-contained apps, set the SelfContained property to true or
use the --self-contained argument. [C:\wix\src\api\burn\test\WixToolsetTest.Mba.Core\WixToolsetTest.Mba.Core.csproj]

The .csproj in question says it's .net 6.0.  VS 17.8.2 only installed the .NET 8 SDK.  I manually installed the .NET 6 SDK and even tried updating the global.json in the wix directory to say version 6.0.417 but I still keep getting this error.

Does anyone know how to resolve this?   After updating the global.json, dotnet --version says 6.0.417 but I still get the build error.

**Comment by chrpai** [comment]
I was able to get past this by just setting the SelfContained property to true but it's still building on .NET 8 which I assume is wrong.  But later I get this error:

C:\GitHub\wix\src\wix\WixToolset.Core.ExtensionCache\WixToolset.Core.ExtensionCache.csproj : error NU1903: Package 'NuGet.Protocol'
6.3.1 has a known high severity vulnerability, https://github.com/advisories/GHSA-6qmf-mmc7-6c2p [C:\GitHub\wix\src\wix\wix.sln]

**Comment by chrpai** [comment]
I was able to get it to build all the way through but it took modifying 5 files.

src\api\burn\test\WixToolsetTest.Mba.Core\WixToolsetTest.Mba.Core.csproj `<SelfContained>true</SelfContained>`
src\ext\Bal\test\examples\examples.proj  remove PublishTrimmed and TrimMode msbuild properties
src\internal\SetBuildNumber\Directory.Packages.props.pp  update Nuget.Protocol to 6.6.1
src\wix\test\WixToolsetTest.Core.Native\WixToolsetTest.Core.Native.csproj  `<SelfContained>true</SelfContained>`
src\wix\test\WixToolsetTest.Sdk\TestData\MultiTargetingWixlib\TestExe\TestExe.csproj  `<SelfContained>true</SelfContained>`

**Comment by barnson** [comment]
https://github.com/wixtoolset/wix/pull/472

**Comment by chrpai** [comment]
@barnson thanks!  I don't have a lot of experience working on WiX so my first step is to make sure I can build it all the way through and confirm that I still can after any changes that I make.  This was an exciting few hours last night working through these issues that you already solved.  I'll resume my efforts when your PR is merged.
